src/abstractdataframe/subset.jl

```diff
@@ -191,7 +215,10 @@ If `ungroup=false` the resulting data frame is re-grouped based on the same
 grouping columns as `gdf` and a `GroupedDataFrame` is returned.
 
 If `GroupedDataFrame` is subsetted then it must include all groups present in the
-`parent` data frame, like in [`select!`](@ref).
+`parent` data frame, like in [`select!`](@ref). Also note that in general in this
+case the parent of the passed `GroupedDataFrame` is mutaded in place, which means
+that the source `GroupedDataFrame` is not safe to use as most likely it will
+get corrupted due to row removal in the parent.
```

Comment:
```suggestion
`parent` data frame, like in [`select!`](@ref). Also note that in general
the parent of the passed `GroupedDataFrame` is mutated in place, which means
that the `GroupedDataFrame` is not safe to use as most likely it will
get corrupted due to row removal in the parent.
```

BTW, maybe we should fix this by updating the `GroupedDataFrame`?